torchgeo/trainers/byol.py

```diff
@@ -390,24 +385,23 @@ def configure_optimizers(self) -> Dict[str, Any]:
             a "lr dict" according to the pytorch lightning documentation --
             https://pytorch-lightning.readthedocs.io/en/latest/common/lightning_module.html#configure-optimizers
         """
-        optimizer_class = getattr(optim, self.hparams.get("optimizer", "Adam"))
-        lr = self.hparams.get("lr", 1e-4)
-        weight_decay = self.hparams.get("weight_decay", 1e-6)
+        hparams = cast(Dict[str, Any], self.hparams)
+        optimizer_class = getattr(optim, hparams.get("optimizer", "Adam"))
+        lr = hparams.get("lr", 1e-4)
+        weight_decay = hparams.get("weight_decay", 1e-6)
         optimizer = optimizer_class(self.parameters(), lr=lr, weight_decay=weight_decay)
 
         return {
             "optimizer": optimizer,
             "lr_scheduler": {
                 "scheduler": ReduceLROnPlateau(
-                    optimizer, patience=self.hparams["learning_rate_schedule_patience"]
+                    optimizer, patience=hparams["learning_rate_schedule_patience"]
                 ),
                 "monitor": "val_loss",
             },
         }
 
-    def training_step(  # type: ignore[override]
-        self, batch: Dict[str, Any], batch_idx: int
-    ) -> Tensor:
+    def training_step(self, *args: Any, **kwargs: Any) -> Tensor:
         """Training step - reports BYOL loss.
 
         Args:
```

Comment:
Docstring doesn't match function signature -- should we copy docstring from lightning here?

Reply:
Will make these more uniform in the next commit.